packages/editor-ui/src/utils/rbacUtils.ts

```diff
@@ -11,20 +11,23 @@ export function inferProjectIdFromRoute(to: RouteLocationNormalized): string {
 
 export function inferResourceTypeFromRoute(to: RouteLocationNormalized): Resource | undefined {
 	const routeParts = to.path.split('/');
-	const routeMap = {
+	const routeMap: Record<string, string> = {
 		workflow: 'workflows',
 		credential: 'credentials',
 		user: 'users',
 		variable: 'variables',
 		sourceControl: 'source-control',
-		externalSecretsStore: 'external-secrets',
+		externalSecret: 'external-secrets',
 	};
 
-	for (const resource of Object.keys(routeMap) as Array<keyof typeof routeMap>) {
-		if (routeParts.includes(routeMap[resource])) {
-			return resource;
+	for (const resource of Object.keys(routeMap)) {
+		const mappedRoute = routeMap[resource];
+		if (routeParts.includes(mappedRoute)) {
+			return resource as Resource;
 		}
 	}
```

Comment:
```suggestion
const isResource = (key: string): key is Resource => key in routeMap;

	for (const key in routeMap) {
		if (isResource(key)) {
			const resource = key;
			const mappedRoute = routeMap[resource];
			if (mappedRoute && routeParts.includes(mappedRoute)) {
				return resource;
			}
		}
	}
```
We could remove coercing by implementing a typeguard:

Reply:
Updated, simplified it as well. Thanks.